samples: sid_end_device: update MCUboot configuration (#778)
nrfconnect/sdk-sidewalk: doc/releases_and_migration/migration_guide_addon_changelog.rst

.. _migration_guide_addon_changelog:

Migration guide for Amazon Sidewalk Add-On v1.2.99

> ###################################################

Reviewer: ```suggestion
Migration guide for Amazon Sidewalk Add-On v1.2.99
##################################################
```

.. _migration_guide_addon_changelog:

Migration guide for Amazon Sidewalk Add-On (TBD)
################################################

.. contents::
   :local:
   :depth: 2

This guide assists you in migrating from Amazon Sidewalk Add-On v1.2.0 to (TBD).

Overview
********

TODO: Add overview of changes.

Migration steps
***************

TODO: Add migration summary.

.. _migration_changelog_mcuboot_rsa:

Keeping RSA MCUboot signatures
******************************

Previous add-on releases forced **RSA** signature type for MCUboot on all platforms (``SB_CONFIG_BOOT_SIGNATURE_TYPE_RSA``).

> This release removes that override and uses the NCS default (**ED25519** on nRF54L Series platforms).

Reviewer: ```suggestion
Earlier add-on releases forced the RSA signature type (``SB_CONFIG_BOOT_SIGNATURE_TYPE_RSA``) for MCUboot across all platforms.
This release removes that override and uses the nRF Connect SDK default (ED25519 on the nRF54L Series platforms).
```

> If your product was developed against a previous add-on version and you must keep using **RSA** for application images signing - for example, to stay compatible with MCUboot on already-deployed devices - add the following to your application's :file:`Kconfig.sysbuild` file:

Reviewer: ```suggestion
If your product was based on a previous add-on version and you must keep using RSA for signing application images.
For example, to stay compatible with MCUboot on already-deployed devices, add the following to your application's :file:`Kconfig.sysbuild` file:
```

Author: "add the following to your application's" applies to the main condition ("If your product was based on a previous add-on version"), not just to the example case, so I think we need to split it differently if we want to avoid long sentences:

How about:
```
If your product was based on a previous add-on version you must keep using RSA for signing application images, for example, to stay compatible with MCUboot on already-deployed devices.
To force RSA signature type in your application, add the following to your application's :file:`Kconfig.sysbuild` file:
```
?

Reviewer: Noted - the updated suggestion sounds good :)

.. code-block:: kconfig

   choice BOOT_SIGNATURE_TYPE
           default BOOT_SIGNATURE_TYPE_RSA
   endchoice

Perform a pristine build of the application after adding this override.

> .. warning::

Reviewer: should be a note
```suggestion
.. note::
```

   A device running MCUboot configured for RSA will not boot ED25519-signed images until MCUboot is updated and re-flashed with a matching configuration.

.. note::

>    ED25519 is the recommended option for new designs on nRF54L Series platforms due to its shorter verification time and smaller key size.

Reviewer: ```suggestion
.. note::
   Use ED25519 for new designs on nRF54L Series platforms because of its shorter verification time and smaller key size.
```
